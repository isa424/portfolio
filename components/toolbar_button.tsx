import { FC, PropsWithChildren } from 'react';

type IToolbarButton = {
	type: 'close' | 'minimize' | 'fullscreen'
};

// todo: Add onClick
export const ToolbarButton: FC<PropsWithChildren<IToolbarButton>> = (props) => {
	const {type} = props;

	const bgTypes: Record<IToolbarButton['type'], string> = {
		'close': 'bg-red-400 hover:bg-red-500',
		'minimize': 'bg-yellow-400 hover:bg-yellow-500',
		'fullscreen': 'bg-green-400 hover:bg-green-500',
	};

	const childBgTypes: Record<IToolbarButton['type'], string> = {
		'close': 'bg-red-400 group-hover:bg-red-700',
		'minimize': 'bg-yellow-400 group-hover:bg-yellow-700',
		'fullscreen': 'bg-green-400 group-hover:bg-green-700',
	};

	return (
		<button
			role={'button'}
			className={`group rounded-full h-3 w-3 ${bgTypes[type]} focus:outline-none cursor-default flex items-center justify-center ring-1 ring-black/10 ring-inset`}
		>
			<div className={`rounded-full ${childBgTypes[type]} h-1 w-1`}></div>
		</button>
	);
};
