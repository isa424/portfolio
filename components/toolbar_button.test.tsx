import { ToolbarButton } from './toolbar_button';
import { render, screen, fireEvent } from '@testing-library/react';

describe('toolbar button', () => {
	it('should render successfully', () => {
		const {container} = render(<ToolbarButton type="close"/>);

		expect(container).toMatchSnapshot();
	});

	it('should render correctly when hovered', () => {
		const {container} = render(<ToolbarButton type={'minimize'}/>)

		const button = screen.getByRole('button');

		fireEvent.mouseOver(button);

		expect(container).toMatchSnapshot();
	});

	it('should have a round div inside on hover', () => {
		const {container} = render(
			<ToolbarButton type={'close'}/>
		);

		const button = screen.getByRole('button');

		fireEvent.mouseOver(button, {});

		expect(container.childElementCount).toEqual(1);

		const child = button.children[0];
		expect(child).toBeInstanceOf(HTMLDivElement);
	});
});
