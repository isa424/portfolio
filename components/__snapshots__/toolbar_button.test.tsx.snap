// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toolbar button should render correctly when hovered 1`] = `
<div>
  <button
    class="group rounded-full h-3 w-3 bg-yellow-400 hover:bg-yellow-500 focus:outline-none cursor-default flex items-center justify-center ring-1 ring-black/10 ring-inset"
    role="button"
  >
    <div
      class="rounded-full bg-yellow-400 group-hover:bg-yellow-700 h-1 w-1"
    />
  </button>
</div>
`;

exports[`toolbar button should render successfully 1`] = `
<div>
  <button
    class="group rounded-full h-3 w-3 bg-red-400 hover:bg-red-500 focus:outline-none cursor-default flex items-center justify-center ring-1 ring-black/10 ring-inset"
    role="button"
  >
    <div
      class="rounded-full bg-red-400 group-hover:bg-red-700 h-1 w-1"
    />
  </button>
</div>
`;
